_posts/2025-07-16-21.0.0-release.md

```diff
@@ -0,0 +1,101 @@
+---
+layout: post
+title: "Apache Arrow 21.0.0 Release"
+date: "2025-07-16 00:00:00"
+author: pmc
+categories: [release]
+---
+<!--
+{% comment %}
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to you under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+The Apache Arrow team is pleased to announce the 21.0.0 release. This release
+covers over 2 months of development work and includes [**1234 resolved
+issues**][1] on [**1234 distinct commits**][2] from [**1234 distinct
+contributors**][2]. See the [Install Page](https://arrow.apache.org/install/) to
+learn how to get the libraries for your platform.
+
+The release notes below are not exhaustive and only expose selected highlights
+of the release. Many other bugfixes and improvements have been made: we refer
+you to the [complete changelog][3].
+
+## Community
+
+Since the 20.0.0 release, Alenka Frim has been invited to join the Project
+Management Committee (PMC).
+
+Thanks for your contributions and participation in the project!
+
+## Release Highlights
+
+## Columnar Format
+
+## Arrow Flight RPC Notes
+
+## C++ Notes
+
+### Compute
+
+### Acero
+
+### Filesystems
+
+### Flight RPC
+
+### Parquet
+
+## C# Notes
+
+- TODO: Note about future repo move plan. "This is the final release of the C# implementation of Arrow as part of the monorepo...etc.
+
+## Java, JavaScript, Go, and Rust Notes
+
+The Java, JavaScript, Go, and Rust Go projects have moved to separate repositories outside
+the main Arrow [monorepo](https://github.com/apache/arrow).
+
+- For notes on the latest release of the [Java
+implementation](https://github.com/apache/arrow-java), see the latest [Arrow
+Java changelog][7].
+- For notes on the latest release of the [JavaScript
+implementation](https://github.com/apache/arrow-js), see the latest [Arrow
+JavaScript changelog][8].
+- For notes on the latest release of the [Rust
+  implementation](https://github.com/apache/arrow-rs) see the latest [Arrow Rust
+  changelog][5].
+- For notes on the latest release of the [Go
+implementation](https://github.com/apache/arrow-go), see the latest [Arrow Go
+changelog][6].
+
+## Linux Packaging Notes
+
+## Python Notes
```

Comment:
@sgilmore10 @kevingurney do either of you want to highlight anything from https://github.com/apache/arrow/issues?q=is%3Aissue%20state%3Aclosed%20label%3A%22Component%3A%20MATLAB%22%20milestone%3A21.0.0 here? We can add a section above the Python header.

Reply:
Yeah, I think there are 1 or 2 issues worth mentioning. I'll work on this.